src/ray/raylet/node_manager.cc

```diff
@@ -1050,6 +1053,24 @@ void NodeManager::ProcessSubmitTaskMessage(const uint8_t *message_data) {
   TaskExecutionSpecification task_execution_spec(
       from_flatbuf<ObjectID>(*message->execution_dependencies()));
   TaskSpecification task_spec(*message->task_spec());
+
+  // Set the current version of the actor that this task is for. This is used
+  // to determine task failure.
+  if (!task_spec.ActorId().IsNil()) {
+    const auto &actor_entry = actor_registry_.find(task_spec.ActorId());
+    int64_t actor_version = 0;
+    // If we have any information about the actor this task is for, use the
+    // latest version of the actor.
+    // TODO(swang): Ideally, we would use the latest global version of the
+    // actor (i.e., look up the actor table entry) if we don't already have it
+    // in our local registry. This should only be a problem for restarted
+    // actors and nodes that have just joined the cluster.
+    if (actor_entry != actor_registry_.end()) {
+      actor_version = actor_entry->second.GetActorVersion();
+    }
+    task_execution_spec.SetActorVersion(actor_version);
```

Comment:
I'm trying to understand the comment above. Is it saying that in some settings (restarted actors and nodes that have just joined the cluster), the `actor_version` that we're setting here can be incorrect? If so, then what is the consequence of an incorrect value?